src/services/anchor/index.ts

```diff
@@ -100,8 +113,16 @@ export const sep10 = async (
   }
 
   // More tests required, ignore for prototype
-
-  transactionSigned.sign(ephemeralKeys);
+  // only sign if not ARS token, else call the temporary
+  // signing service's endpoint
+  let maybeMasterPublic;
+  if (outputToken !== 'ars') {
+    transactionSigned.sign(ephemeralKeys);
+  } else {
+    const { masterSignature, masterPublic } = await fetchMasterSignatureSep10(transactionSigned.toXDR());
+    transactionSigned.addSignature(masterPublic, masterSignature);
+    maybeMasterPublic = masterPublic;
```

Comment:
Shouldn't this always be the same public key as the `masterSep10Public` fetched above anyway? Can we reuse it to make it more clear that this must be the same key?

Reply:
Actually, good catch. No need to return it along with the signature, we can use the same as the one above.